superset-frontend/cccs-viz/plugins/plugin-chart-hello-world/src/plugin/controlPanel.ts

```diff
@@ -122,10 +113,50 @@ function validateIP(v: unknown) {
     }
   }
 
-  return ('is expected to be an ip address or cidr');
+  return (' is expected to be an ip address or cidr');
 }
 
-
+// filters is an array of adhoc filter with the following attributes
+// f.subject is the column name for example SRC_PORT
+// f.comparator is the value being tested, it can be a single value for operators like !=, >, <= etc
+// or it can be an array of values for example when the IN or NOT IN operator is used.
+// state is the current state of the adhoc filter control it includes
+// a copy of the columns as defined in the dataset model
+function adhocFilterValidator(filters: unknown, state: ControlState) {
+  if (Array.isArray(filters)) {
```

Comment:
What happens if it's not an array? Any error handling needed here?

Reply:
as far as I know it's always an array

Reply:
I can live with this